src/components/DonatePage/DonatePageBody.tsx

```diff
@@ -499,18 +501,38 @@ const DonatePageBody: React.FC<DonatePageBodyProps> = ({ setIsConfirming }) => {
         setInputBalanceError(false);
         return null;
       } else {
-        const amount = floor((userDonationCap * percentage) / 100);
-        setInputAmount(amount.toString());
-        if (amount > parseFloat(tokenDetails.formattedBalance)) {
+        const remainingBalance = floor(tokenDetails?.formattedBalance);
+        const amount = floor((remainingBalance * percentage) / 100);
+        if (amount <= userDonationCap) {
+          setInputAmount(amount.toString());
+        } else if (amount > userDonationCap) {
+          setInputAmount(userDonationCap.toString());
+        }
```

Comment:
```suggestion
        setInputAmount(Math.min(amount, userDonationCap).toString());
```

Reply:
Thanks @aminlatifi i have updated the changes